x/reward/client/cli/cli_test.go

```diff
@@ -0,0 +1,1023 @@
+				fmt.Sprintf("--max-reward-by-address=100%s", s.cfg.BondDenom),
+				"--claim-periods=52",
+				"--claim-period-days=abc",
+				fmt.Sprintf("--start-time=%s", soon.Format(time.RFC3339)),
+				"--expire-days=14",
+				fmt.Sprintf("--qualifying-actions=%s", "actions"),
+			},
+			true,
+			"invalid argument \"abc\" for \"--claim-period-days\" flag: strconv.ParseUint: parsing \"abc\": invalid syntax",
+			0,
+		},
+		{"add reward program tx - invalid type for claim period days",
+			[]string{
+				"test add reward program",
+				"description",
+				fmt.Sprintf("--total-reward-pool=580%s", s.cfg.BondDenom),
+				fmt.Sprintf("--max-reward-by-address=100%s", s.cfg.BondDenom),
+				"--claim-periods=52",
+				"--claim-period-days=10",
+				fmt.Sprintf("--start-time=%s", soon.Format(time.RFC3339)),
+				"--expire-days=abc",
+				fmt.Sprintf("--qualifying-actions=%s", "actions"),
+			},
+			true,
+			"invalid argument \"abc\" for \"--expire-days\" flag: strconv.ParseUint: parsing \"abc\": invalid syntax",
+			0,
+		},
+		{"add reward program tx - invalid type for claim period days",
+			[]string{
+				"test add reward program",
+				"description",
+				fmt.Sprintf("--total-reward-pool=580%s", s.cfg.BondDenom),
+				fmt.Sprintf("--max-reward-by-address=100%s", s.cfg.BondDenom),
+				"--claim-periods=52",
+				"--claim-period-days=10",
+				fmt.Sprintf("--start-time=%s", soon.Format(time.RFC3339)),
+				"--expire-days=14",
+				"--max-rollover-periods=abc",
+				fmt.Sprintf("--qualifying-actions=%s", "actions"),
+			},
+			true,
+			"invalid argument \"abc\" for \"--max-rollover-periods\" flag: strconv.ParseUint: parsing \"abc\": invalid syntax",
+			0,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			clientCtx := s.network.Validators[0].ClientCtx
+			args := []string{
+				fmt.Sprintf("--%s=%s", flags.FlagFrom, s.network.Validators[0].Address.String()),
+				fmt.Sprintf("--%s=true", flags.FlagSkipConfirmation),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(10))).String()),
+			}
+			tc.args = append(tc.args, args...)
+
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, rewardcli.GetCmdRewardProgramAdd(), append(tc.args, []string{fmt.Sprintf("--%s=json", tmcli.OutputFlag)}...))
+			var response sdk.TxResponse
+			marshalErr := clientCtx.Codec.UnmarshalJSON(out.Bytes(), &response)
+			if tc.expectErr {
+				s.Assert().Error(err)
+				s.Assert().Equal(tc.expectErrMsg, err.Error())
+			} else {
+				s.Assert().NoError(err)
+				s.Assert().NoError(marshalErr, out.String())
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestTxClaimReward() {
+	testCases := []struct {
+		name         string
+		args         []string
+		expectErr    bool
+		expectErrMsg string
+		expectedCode uint32
+	}{
+		{"claim rewards tx - valid",
+			[]string{
+				"1",
+			},
+			false,
+			"",
+			0,
+		},
```

Comment:
Yep, good catch.  We were doing a lot of copy and paste for our tests.  Passing in a []string or 1 argument doesn't make sense in these cases.  Also, I will keep an eye out for patterns like this in other modules (since they were a copy and paste).  Thanks, good catches.